lxd/storage/drivers/driver_pure.go

```diff
@@ -283,7 +283,34 @@ func (d *pure) Unmount() (bool, error) {
 
 // GetResources returns the pool resource usage information.
 func (d *pure) GetResources() (*api.ResourcesStoragePool, error) {
+	pool, err := d.client().getStoragePool(d.name)
+	if err != nil {
+		return nil, err
+	}
+
 	res := &api.ResourcesStoragePool{}
+
+	res.Space.Total = uint64(pool.Quota)
+	res.Space.Used = uint64(pool.Space.UsedBytes)
+
+	if pool.Quota == 0 {
```

Comment:
@hamistao is this aligned with how we are going to report storage pool space for unbounded pools (like dir)? 

Is it just unbounded volumes which we will report as 0 size?

Reply:
Currently, if the storage pool is unbound, the entire storage space is reported. Let me know what is desired outcome in such case. Probably just `0`?

Reply:
We had decided to use `-1` in such cases. The PR setting this standard wasn't merged yet so we have time to change if needed, but personally I prefer `-1`.

Reply:
-1 for unbounded volume size is ok, but i dont think -1 for unbounded pools is a good idea, i thought we had agreed to present the entire capacity of the underlying storage device so that it was possible to ascertain space for a particular unbounded volume by considering the pool capacity?